x-pack/plugins/ml/public/services/results_service.js

```diff
@@ -1452,25 +1451,40 @@ function getEventDistributionData(
 
     const body = {
       query: {
-        bool: {
-          must: mustCriteria
+        // using function_score and random_score to get a random sample of documents.
+        // otherwise all documents would have the same score and the sampler aggregation
+        // would pick the first N documents instead of a random set.
+        function_score: {
+          query: {
+            bool: {
+              must: mustCriteria
+            }
+          },
+          functions: [
+            {
+              random_score: {
+                // static seed to get same randomized results on every request
+                seed: 'ml'
```

Comment:
Looks like you now need to use a `field` parameter here as well as `seed` to get reproducible results - see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html#function-random

Reply:
Thanks for pointing that out, I was working off some outdated docs. Push an update in 43ced43186.